### DISCUSSION TITLE
list empty verification sql statement

### DISCUSSION THREAD

**Discussion by lfoIncluit** [lf-edge/ekuiper#1391] [Q&A] [answered]
How can I verify if list is empty in a rule statement. Example:

`
{
    "sql": "SELECT at1,at2,at3 FROM stream1 WHERE at1 != [] AND at1[:]->subat1 != NULL",
    "actions": [
        {
            "log": {}
        },
        {
            "mqtt": {
                "server": "tcp://172.17.0.1:1883",
                "topic": "topic1"
            }
        }
    ]
}
`

**Comment by ngjaying** [comment, marked as answer]
There is a cardinality function to check the length of a list: `cardinality(at1) = 0`. Please check the doc https://ekuiper.org/docs/en/latest/sqls/built-in_functions.html#other-functions
